compiler/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -194,6 +194,17 @@ impl<'interner> TypeChecker<'interner> {
                     return Type::Error;
                 }
 
+                // Check that we are not passing a reference from an unconstrained runtime to a constrained runtime
+                if is_current_func_constrained
+                    && is_unconstrained_call
+                    && matches!(&return_type.follow_bindings(), Type::MutableReference(_))
```

Comment:
We could probably combine this with the if statement above that also checks `is_current_func_constrained` and `is_unconstrained_call`

Reply:
I have combined these if statements to be a single if statement.